lmorpho/quaternion.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <cmath>
+
+struct quaternion {
+    double w, x, y, z;
```

Comment:
This needs a link to the notation that is being used. I know they are black boxes, but to validate what is going on.